mods/ra/maps/sarin-gas-2-down-under/downunder.lua

```diff
@@ -0,0 +1,457 @@
+--[[
+   Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+SarinDispensed = false
+TanyaFreed = false
+TruckStolen = false
+SovietImportantGuys = { Officer3, Officer2, Officer1, Scientist1, Scientist2 }
+Camera1Towers = { FlameTower2, FlameTower3 }
+TruckExit = { TruckDrive1, TruckDrive2 }
+TruckEntry = { TruckDrive3.Location, TruckDrive4.Location, TruckDrive5.Location }
+TruckType = { "truk" }
+DogCrew = { DogCrew1, DogCrew2, DogCrew3 }
+SarinVictims = { SarinVictim1, SarinVictim2, SarinVictim3, SarinVictim4, SarinVictim5, SarinVictim6, SarinVictim7, SarinVictim8, SarinVictim9, SarinVictim10, SarinVictim11, SarinVictim12, SarinVictim13 }
+Camera2Team = { Camera2Rifle1, Camera2Rifle2, Camera2Rifle3, Camera2Gren1, Camera2Gren2 }
+PrisonAlarm = { CPos.New(55,75), CPos.New(55,76), CPos.New(55,77), CPos.New(55,81), CPos.New(55,82), CPos.New(55,83), CPos.New(60,77), CPos.New(60,81), CPos.New(60,82), CPos.New(60,83) }
+GuardDogs = { PrisonDog1, PrisonDog2, PrisonDog3, PrisonDog4 }
+TanyaTowers = { FlameTowerTanya1, FlameTowerTanya2 }
+ExecutionArea = { CPos.New(91, 70), CPos.New(92, 70), CPos.New(93, 70), CPos.New(94, 70) }
+FiringSquad = { FiringSquad1, FiringSquad2, FiringSquad3, FiringSquad4, FiringSquad5, Officer2 }
+DemoTeam = { DemoDog, DemoRifle1, DemoRifle2, DemoRifle3, DemoRifle4, DemoFlame }
+DemoTruckPath = { DemoDrive1, DemoDrive2, DemoDrive3, DemoDrive4 }
+WinTriggerArea = { CPos.New(111, 59), CPos.New(111, 60), CPos.New(111, 61), CPos.New(111, 62), CPos.New(111, 63), CPos.New(111, 64), CPos.New(111, 65) }
+
+ObjectiveTriggers = function()
+	Trigger.OnEnteredFootprint(WinTriggerArea, function(a, id)
+		if not EscapeGoalTrigger and a.Owner == greece then
+			EscapeGoalTrigger = true
+			greece.MarkCompletedObjective(ExitBase)
+			if Map.LobbyOption("difficulty") == "hard" then
+				greece.MarkCompletedObjective(NoCasualties)
+			end
+			if TanyaFreed == false then
+				greece.MarkFailedObjective(FreeTanya)
+			else
+				greece.MarkCompletedObjective(FreeTanya)
+			end
+		end
+	end)
+
+	Trigger.OnKilled(Tanya, function()
+		greece.MarkFailedObjective(FreeTanya)
+	end)
+
+	Trigger.OnAllKilled(TanyaTowers, function()
+		TanyaFreed = true
+	end)
+
+	Trigger.OnAllKilled(SovietImportantGuys, function()
+		greece.MarkCompletedObjective(KillVIPs)
+	end)
+
+	Trigger.OnInfiltrated(WarFactory2, function()
+		if not StealMammoth.IsDead or StealMammoth.Owner ~= ussr then
```

Comment:
Doesn't this want to be `not StealMammoth.IsDead and StealMammoth.Owner == ussr`?